Assignments/Program.cs

```diff
@@ -11,34 +11,39 @@
 using System.Linq;
 using System.IO;
 using System.Diagnostics;
+using System.Reflection;
 
 namespace Academy;
 public class A14 {
    public static void Main () {
-      var wordList = new List<string> (File.ReadAllLines (@"c:/etc/words.txt"));
+      // Loading all the dictionary words from a assembly-manifest resource file.
+      using var reader = new StreamReader (Assembly.GetExecutingAssembly ().GetManifestResourceStream ($"Assignments.Files.words.txt"));
+      var wordList = reader.ReadToEnd ().Split ("\r\n");
       Dictionary<string, List<string>> anagrams = new ();
       int count = 0;
       int[] hashTable = new int[26];
       var watch = Stopwatch.StartNew (); //.............................TIMER ON
       foreach (string s in wordList) {
-         string Hash = HashCode (ref hashTable, s);
-         if (anagrams.TryGetValue (Hash, out List<string> value)) value.Add (s);
-         else anagrams.Add (Hash, new List<string> { s });
+         string hash = HashCode (ref hashTable, s);
+         if (anagrams.TryGetValue (hash, out List<string> value)) value.Add (s);
+         else anagrams.Add (hash, new List<string> { s });
       }
       watch.Stop ();//..................................................TIMER OFF
       Console.WriteLine ("\x1B[4m" + "Anagrams:-" + "\x1B[0m");
       foreach (var kvp in anagrams.Where (a => a.Value.Count > 1).OrderByDescending (a => a.Value.Count))
          Console.WriteLine ($"{++count,4}. {kvp.Value.Count,2}- {string.Join (", ", kvp.Value)}");
       Console.WriteLine ($"\nThe Execution time of the program: {watch.ElapsedMilliseconds} ms");
-
-      /// <summary>Returns an unique string made of hash-table digits of given input string</summary>
-      static string HashCode (ref int[] array, string word) {
-         array = new int [26];
-         foreach (char c in word) array[c - 65]++;
-         return string.Join ("", array);
-      }
    }
 
+   /// <summary>Returns an unique string made of hash-table digits of given input string</summary>
+   // Basically, this method returns a string whose length is 26 with chars either '1' or '0'.
+   // The char at Nth index in the returned string represents the presence/absence of (N+1)th alphabet in the given word given word.
+   // So the char '1' at 0th index in the returned string implies that 1st alphabet 'A' is present in the given string & vice versa.
+   static string HashCode (ref int[] array, string word) {

```

Comment:
Please explain why use ref in your most function parameters.

Reply:
This way we can reuse the same array created in the Main method, many times.

Reply:
Please see the answer from Jon Skeet: https://stackoverflow.com/questions/635915/when-to-use-ref-and-when-it-is-not-necessary-in-c-sharp